low-level/files/opt/seagate/sspl/setup/sspl_setup_init.py

```diff
@@ -0,0 +1,201 @@
+            print(f"Checking for required processes running state for role '{role}'")
+            self.check_for_active_processes(self.SSU_REQUIRED_PROCESSES)
+        elif role == "vm" or role == "gw" or role == "cmu":
+            print(f"Checking for dependency rpms for role '{role}'")
+            # No dependency currently. Keeping this section as it may be
+            # needed in future.
+            self.check_for_dep_rpms(self.VM_DEPENDENCY_RPMS)
+            # No processes to check in vm env
+        else:
+            print(f"No rpm or process dependencies set, to check for supplied role {role}, skipping checks.\n")
+
+    def get_uid(self, proc_name : str) -> int:
+        uid = -1
+        try :
+            uid = pwd.getpwnam(proc_name).pw_uid
+        except KeyError :
+            print(f"No such User: {proc_name}")
+        return uid
+
+    def process(self):
+        if not self.args:
+            self.usage()
+
+        i = 0
+        while i < len(self.args):
+            if self.args[i] == '-dp':
+                # Extract the data path
+                sspldp = ''
+                with open(file_store_config_path, mode='rt') as confile:
+                    for line in confile:
+                        if line.find('data_path') != -1:
+                            sspldp = line.split('=')[1]
+                            break
+                # Crete the directory and assign permissions
+                try:
+                    os.makedirs(sspldp, mode=0o766, exist_ok=True)
+                    sspl_ll_uid = self.get_uid('sspl-ll')
+                    if sspl_ll_uid == -1:
+                        sys.exit(1)
+                    os.chown(sspldp, sspl_ll_uid, -1)
+                    for root, dirs, files in os.walk(sspldp):
+                        for item in dirs:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                        for item in files:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                except OSError as error:
+                    print(error)
+                    sys.exit(1)
+
+            elif self.args[i] == '-r':
+                i+=1
+                if i>= len(self.args) or self.args[i] not in roles:
+                    self.usage()
+                else:
+                    self.role = self.args[i]
+            else:
+                self.usage()
+            i+=1
+        
+        # Create /tmp/dcs/hpi if required. Not needed for '<product>' role
+        if self.role != "cortx":
+            try:
+                os.makedirs(self.hpi_path, mode=0o777, exist_ok=True)
+                zabbix_uid = self.get_uid('zabbix')
+                if zabbix_uid != -1:
+                    os.chown(self.hpi_path, zabbix_uid, -1)
+            except OSError as error:
+                print(error)
+
+        # Check for sspl required processes and misc dependencies like installation, etc based on 'role'
+        if self.role:
```

Comment:
Is that the correct conversion for the below code?
'''
[ -z "$role" ] || check_dependencies $role
'''
I am seeing or operator inside to statement. If 1st true then doesn't go to the 2nd one.
But you are going inside the condition when [ -z "$role" ] is true.
Please recheck all the conditions again.
FYI - @cdeshmukh

Reply:
if role is not available or None then first condition will be true and it won't go to next condition, but it role is not zero or None i.e if role is set then check for dependencies.